### PR TITLE
Add symmetric quantization with no clipping error in the tensor subclass based API

### PR DESCRIPTION
**Pull request by iseeyuan**
Follow up with #805 , add the same option to the new tensor subclass API.

### CODE REVIEW

File: test/quantization/test_quant_api.py [hunk 1]
```diff
@@ -526,6 +526,31 @@ def test_quantized_tensor_subclass_8da4w(self):
...
+
+        quantizer = Int8DynActInt4WeightQuantizer(groupsize=group_size, mapping_type=MappingType.SYMMETRIC_NO_CLIPPING_ERR)
+        m_copy = quantizer.quantize(m_copy)
+        assert isinstance(m_copy.linear1, Int8DynActInt4WeightLinear)
+        assert isinstance(m_copy.linear2, Int8DynActInt4WeightLinear)
+
+        res = m(*example_inputs)
+        ref = m_copy(*example_inputs)
+        self.assertTrue(torch.equal(res, ref))
+    @unittest.skipIf(not TORCH_VERSION_AT_LEAST_2_4, "Test only enabled for 2.4+")
```
Comment (jerryzh168) [inline test/quantization/test_quant_api.py:553]: why is this changed?

File: test/quantization/test_quant_api.py [hunk 2]
```diff
@@ -526,6 +526,31 @@ def test_quantized_tensor_subclass_8da4w(self):
         self.assertTrue(torch.equal(res, ref))
 
     @unittest.skipIf(not TORCH_VERSION_AT_LEAST_2_4, "Test only enabled for 2.4+")
+    def test_quantized_tensor_subclass_8da4w_no_clipping_err(self):
```
Comment (jerryzh168) [inline test/quantization/test_quant_api.py:529]: maybe this can be merged with the previous test, using https://github.com/pytorch/ao/blob/e1039abac7f429a8d7f489d047d9b34d6ac6afe2/test/dtypes/test_affine_quantized_float.py#L52

File: test/quantization/test_quant_api.py (hunk 1 above)
Comment (iseeyuan) [reply test/quantization/test_quant_api.py:553]: It may be done by mistake. Let me remove it.

File: torchao/quantization/quant_api.py [hunk 3]
```diff
@@ -473,14 +473,13 @@ def _int8_asymm_per_token_quant(x: torch.Tensor) -> torch.Tensor:
     target_dtype = torch.int8
     return to_affine_quantized_intx(x, mapping_type, _get_per_token_block_size(x), target_dtype)
 
-def apply_int8_dynamic_activation_int4_weight_quant(weight, group_size=32):
+def apply_int8_dynamic_activation_int4_weight_quant(weight, group_size=32, mapping_type=MappingType.SYMMETRIC):
```
Comment (jerryzh168) [inline torchao/quantization/quant_api.py:476]: might be better to clarify this is `weight_mapping_type` I feel

**Comment by jerryzh168** [comment]
@iseeyuan although the merging is not blocked, please still wait for an approval before merging next time :)
